<launch>

   <arg name="server" default="localhost"/>

  <node pkg="vrpn_client_ros" type="vrpn_client_node" name="vrpn_client_node">
    <rosparam subst_value="true">
      server: 192.168.1.105
      port: 3883

      update_frequency: 100.0
      frame_id: world

      # Use the VRPN server's time, or the client's ROS time.
      use_server_time: false
      broadcast_tf: true

      # Must either specify refresh frequency > 0.0, or a list of trackers to create
      refresh_tracker_frequency: 1.0
      trackers:
      - three
      - wl_r
      - wl_o
      - wa_r
      - wa_o
      #- SecondTracker
    </rosparam>
    <remap from="/vrpn_client_node/three/pose" to="/mavros/vision_pose/pose"/>
    <remap from="/vrpn_client_node/wl_r/pose" to="/world_to_link_ori/pose"/>
    <remap from="/vrpn_client_node/wa_r/pose" to="/world_to_apriltags_ori/pose"/>
    <remap from="/vrpn_client_node/wl_o/pose" to="/world_to_link_pose/pose"/>
    <remap from="/vrpn_client_node/wa_o/pose" to="/world_to_apriltags_pose/pose"/>
  </node>

  <node  pkg="apriltags_ros" type="apriltag_detector_node" name="apriltag_detector">
    <rosparam param="tag_descriptions">[
      {id: 0, size: 0.528000},
      {id: 1, size: 0.160000, frame_id: target1},
      {id: 2, size: 0.528000},
      {id: 3, size: 0.528000},
      {id: 4, size: 0.528000},
      {id: 5, size: 0.528000},
      {id: 6, size: 0.528000},
      {id: 7, size: 0.528000},
      {id: 8, size: 0.528000},
      {id: 9, size: 0.528000},
      {id: 10, size: 0.528000},
      {id: 11, size: 0.528000},
      {id: 12, size: 0.528000},
      {id: 13, size: 0.528000},
      {id: 14, size: 0.528000},
      {id: 15, size: 0.528000},
      {id: 16, size: 0.528000},
      {id: 17, size: 0.528000},
      {id: 18, size: 0.528000},
      {id: 19, size: 0.528000},
      {id: 20, size: 0.528000},
      {id: 21, size: 0.528000},
      {id: 22, size: 0.528000},
      {id: 23, size: 0.528000},
      {id: 24, size: 0.528000},
      {id: 25, size: 0.528000},
      {id: 26, size: 0.528000},
      {id: 27, size: 0.528000},
      {id: 28, size: 0.528000},
      {id: 29, size: 0.528000},
      {id: 30, size: 0.528000},
      {id: 31, size: 0.528000},
      {id: 32, size: 0.528000},
      {id: 33, size: 0.528000},
      {id: 34, size: 0.528000},
      {id: 35, size: 0.528000},
      {id: 36, size: 0.528000},
      {id: 37, size: 0.528000},
      {id: 38, size: 0.528000},
      {id: 39, size: 0.528000}]
    </rosparam>
  </node>

  <node ns="camera" pkg="uvc_camera" type="uvc_camera_node" name="uvc_camera" output="screen">
    <param name="width" type="int" value="640" />
    <param name="height" type="int" value="480" />
    <param name="format" type="string" value="jpeg" />
    <param name="fps" type="int" value="60" />
    <param name="frame" type="string" value="webcam" />
    <param name="device" type="string" value="/dev/video0" />
    <param name="camera_info_url" type="string" value="file://$(find ffcalib)/ost_c.yaml" />
    <remap from="/camera/camera_info" to="/f_camera/camera_info"/>
  </node>

  <node pkg="image_transport" name="republish_uvc"  type="republish" args="compressed in:=/camera/image_raw _image_transport:=compressed raw out:=/f_camera/image_raw" />
  <node ns="f_camera"  pkg="image_proc" name="image_proc"  type="image_proc" />

  <node pkg="ffcalib" name="ffcalib"  type="ffcalib" output="screen" />



</launch>
